### ISSUE TITLE
[SR-12080] readabilityHandler on pipe that is Process.standardError sometimes doesn't get called on EOF

### ISSUE THREAD

**Issue by weissi** [apple/swift-corelibs-foundation#3275]

**Comment by beefon** [comment]
Indeed, there is a race between Process termination and the data to be delivered through its pipes. The behaviour clearly differs on macOS and Linux: on macOS there is a final call with empty data, on Linux there isn't. 

I double the last question for advice.

**Comment by eaigner** [comment]
`on macOS there is a final call with empty data, on Linux there isn't.`

No, I am also not seeing a final call with empty data on macOS.
